inc/updater/namespace.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace FAIR\Updater;
+
+use Fragen\Git_Updater;
+use stdClass;
+
+/**
+ * Exit if called directly.
+ */
+if ( ! defined( 'WPINC' ) ) {
+	die;
+}
+
+function bootstrap() {
+	add_action( 'init', __NAMESPACE__ . '\\init' );
+}
+
+/**
+ * Gather all plugins/themes with data in Update URI and DID header.
+ *
+ * @return stdClass
+ */
+function init() {
+	/** @var array */
+	$package_arr = [];
+
+	// Seems to be required for PHPUnit testing on GitHub workflow.
+	if ( ! function_exists( 'get_plugins' ) ) {
+		require_once ABSPATH . 'wp-admin/includes/plugin.php';
+	}
+
+	$plugin_path = trailingslashit( WP_PLUGIN_DIR );
+	$plugins     = get_plugins();
+	foreach ( $plugins as $file => $plugin ) {
+		$update_uri = $plugin['UpdateURI'];
+		$plugin_id  = get_file_data( $plugin_path . $file, [ 'PluginID' => 'Plugin ID' ] )['PluginID'];
+
+		if ( ! empty( $update_uri ) && ! empty( $plugin_id ) ) {
```

Comment:
Can probably take this `! empty( $update_uri )` to before the `get_file_data()` call to save checking the filesystem unnecessarily. Same for the themes.

Reply:
Fixed.